ABB.SrcML.Test/LibSrcMLRunnerTests.cs

```diff
@@ -0,0 +1,481 @@
+    class LibSrcMLRunnerTests {
+        private const string TestInputPath = @"..\..\TestInputs";
+        [TearDown]
+        public void Cleanup() {
+            File.Delete("output0.cpp.xml");
+            File.Delete("output1.cpp.xml");
+        }
+        /// <summary>
+        /// This tests the creation of an archive using a list of source files
+        /// </summary>
+        [Test]
+        public void TestCreateSrcMLArchiveFtF() {
+            LibSrcMLRunner.SourceData ad = new LibSrcMLRunner.SourceData();
+            ad.SetArchiveFilename(Path.Combine(TestInputPath, "input.cpp"));
+            ad.SetArchiveLanguage(LibSrcMLRunner.SrcMLOptions.SRCML_LANGUAGE_CXX);
+
+            LibSrcMLRunner.SourceData bc = new LibSrcMLRunner.SourceData();
+            bc.SetArchiveFilename(Path.Combine(TestInputPath, "input2.cpp"));
+            bc.SetArchiveLanguage(LibSrcMLRunner.SrcMLOptions.SRCML_LANGUAGE_CXX);
+
+            List<LibSrcMLRunner.SourceData> data = new List<LibSrcMLRunner.SourceData>();
+            data.Add(ad);
+            data.Add(bc);
+
+            IntPtr structPtr = LibSrcMLRunner.CreatePtrFromStruct(ad);
+            IntPtr structPtr2 = LibSrcMLRunner.CreatePtrFromStruct(bc);
+
+            List<IntPtr> structArrayPtr = new List<IntPtr>();
+            structArrayPtr.Add(structPtr);
+            structArrayPtr.Add(structPtr2);
+            try {
+                Assert.True(LibSrcMLRunner.SrcmlCreateArchiveFtF(structArrayPtr.ToArray(), structArrayPtr.Count(), "output") == 0);
+            }
+            catch (SrcMLException e) {
+                Console.WriteLine(e.Message);
+            }
+
+            {
+                Assert.True(File.Exists("output0.cpp.xml"));
+                SrcMLFile srcFile = new SrcMLFile("output0.cpp.xml");
+                Assert.IsNotNull(srcFile);
+
+                var files = srcFile.FileUnits.ToList();
+                Assert.AreEqual(1, files.Count());
+
+                string file = Path.Combine(TestInputPath, "input.cpp");
+                var f1 = (from ele in files
+                          where ele.Attribute("filename").Value == file
+                          select ele);
+                Assert.AreEqual(file, f1.FirstOrDefault().Attribute("filename").Value);
+            }
+            {
+                Assert.True(File.Exists("output1.cpp.xml"));
+                SrcMLFile srcFile = new SrcMLFile("output1.cpp.xml");
+                Assert.IsNotNull(srcFile);
+
+                var files = srcFile.FileUnits.ToList();
+                Assert.AreEqual(1, files.Count());
+
+                string file1 = Path.Combine(TestInputPath, "input2.cpp");
+                var f2 = (from ele in files
+                          where ele.Attribute("filename").Value == file1
+                          select ele);
+                Assert.AreEqual(file1, f2.FirstOrDefault().Attribute("filename").Value);
+            }
+            ad.Dispose();
+            bc.Dispose();
```

Comment:
Instead of manually calling Dispose(), you should consider using a using block (https://msdn.microsoft.com/en-us/library/yh598w02.aspx). This is nice because it will automatically handle exceptions properly to make sure Dispose is always called.